src/chart/components/y_axis/label-color.functions.ts

```diff
@@ -4,10 +4,37 @@
  * If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
  */
 import { PriceMovement } from '../../model/candle-series.model';
-import { FullChartColors, YAxisLabelsColors } from '../../chart.config';
+import { FullChartColors, YAxisConfig, YAxisLabelsColors } from '../../chart.config';
+import { getLabelTextColorByBackgroundColor } from '../../utils/canvas/canvas-text-functions.utils';
 
 export const DEFAULT_LABEL_COLOR = '#FF00FF';
 
+export function getPlainLabelTextColor(
+	colorsConfig: FullChartColors,
+	textColor: string,
+	invertedTextColor: string,
+	yAxisState: YAxisConfig,
+) {
+	// `plain` label is transparent, so to calculate text color
+	// we need to go down to draw hierarchy
+	// the next is YAxis bg color
+	const yAxisBGColor = colorsConfig.yAxis.backgroundColor;
+
+	// if yAxis bg color is transparent, then we need to check chart area background color
+	const plainChartBGColor = colorsConfig.chartAreaTheme.backgroundColor;
+	// when chart area bg color is gradient, then we need to check which side yAxis is drawn
+	// because color on the right side and left side is different
+	const leftChartBGColor = colorsConfig.chartAreaTheme.backgroundGradientTopColor;
+	const rightChartBGColor = colorsConfig.chartAreaTheme.backgroundGradientBottomColor;
+	const gradientChartBGColor = yAxisState.align === 'left' ? leftChartBGColor : rightChartBGColor;
```

Comment:
we need to keep in mind, that in case of multiple axis this approximation might work badly